lang/step.md

```diff
@@ -360,18 +360,22 @@ impl<M: Memory> Machine<M> {
         let mut locals: Map<LocalName, Place<M>> = Map::new();
 
         // First evaluate the return place and remember it for `Return`. (Left-to-right!)
-        let (caller_ret_place, caller_ret_abi) = ret;
-        let caller_ret_place = self.eval_place(caller_ret_place)?;
-        // Create place for return local, if needed.
-        if let Some((ret_local, callee_ret_abi)) = func.ret {
-            let callee_ret_layout = func.locals[ret_local].layout::<M>();
-            locals.insert(ret_local, self.mem.allocate(callee_ret_layout.size, callee_ret_layout.align)?);
-            if caller_ret_abi != callee_ret_abi {
-                throw_ub!("call ABI violation: return ABI does not agree");
+        let ret_place = ret.try_map(|(caller_ret_place, caller_ret_abi)| {
+            let caller_ret_place = self.eval_place(caller_ret_place)?;
+
+            // Create place for return local, if needed.
+            if let Some((ret_local, callee_ret_abi)) = func.ret {
+                let callee_ret_layout = func.locals[ret_local].layout::<M>();
+                locals.insert(ret_local, self.mem.allocate(callee_ret_layout.size, callee_ret_layout.align)?);
+                if caller_ret_abi != callee_ret_abi {
+                    throw_ub!("call ABI violation: return ABI does not agree");
+                }
+            } else {
+                // FIXME: Can we truly accept any caller ABI if the callee does not return anything?
             }
-        } else {
-            // FIXME: Can we truly accept any caller ABI if the callee does not return anything?
-        }
+
+            return caller_ret_place;
```

Comment:
This `return caller_ret_place;` instead of `caller_ret_place`,
because `specr` otherwise doesn't wrap it with `specr::hidden::ret(_)`.

(wrapping closure bodies in `ret` causes a few type inference problems, sadly)